platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -1089,24 +1098,56 @@ public void removeAllMarkers() {
      */
     public void setMapChangeListener(@Nullable final MapChangeListener listener) {
         mapChangeListener = listener;
+        mapRegionChangeStateManager = new MapRegionChangeStateManager(mapChangeListener);
     }
 
-    //Convenience member functions
-    private void onRegionWillChange(boolean animated) {
-        if (mapChangeListener != null) {
-            mapChangeListener.onRegionWillChange(animated);
+    private static class MapRegionChangeStateManager {
+
+        enum MapRegionChangeState {
+            IDLE,
+            JUMPING,
+            ANIMATING,
         }
-    }
 
-    private void onRegionDidChange(boolean animated) {
-        if (mapChangeListener != null) {
-            mapChangeListener.onRegionDidChange(animated);
+        WeakReference<MapChangeListener> mapChangeListenerWeakReference;
```

Comment:
Why is this a weak reference?

Reply:
To me it didn't make sense for MapRegionChangeManager to hold a strong reference to MapChangeListener. Also during MapController init, we need to initialize MapRegionChangeManager with a `null`!

Reply:
Also not an issue if we remove the inner class..

Reply:
Inner class is gone => weak reference is gone!